File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/billing/shared/bill-invoice-return-items.model.ts
import * as moment from "moment";

export class BillInvoiceReturnItemsModel {
    public BillReturnItemId: number = 0;
    public BillReturnId: number = 0; //this is FK, so can't have null value
    public BillingTransactionItemId: number = 0; //this is FK, so can't have null value
    public BillingTransactionId: number = 0; //this is FK, so can't have null value
    public PatientId: number = 0; //this is FK, so can't have null value
    public ServiceDepartmentId: number = 0; //this is FK, so can't have null value
    public ServiceItemId: number = 0; //this is FK, so can't have null value
    public ItemName: string = "";
    public Price: number = 0;
    public RetQuantity: number = 0;
    public RetSubTotal: number = 0;
    public RetDiscountAmount: number = 0;
    public RetTaxAmount: number = 0;
    public RetTotalAmount: number = 0;
    public RetDiscountPercent: number = 0;
    public PrescriberId: number = null;
    public PerformerId: number = null;
    public BillStatus: string = null;
    public RequisitionId: number = null;
    public RequisitionDate: string = null;
    public RetCounterId: number = null;
    public RetRemarks: string = null;
    public RequestedBy: number = null;
    public PatientVisitId: number = null;
    public BillingPackageId: number = null;
    public CreatedBy: number = 0;//this is FK, so can't have null value
    public CreatedOn: string = null;
    public BillingType: string = null;
    public RequestingDeptId: number = null;
    public VisitType: string = null;
    public PriceCategory: string = null;
    public PriceCategoryId: number = 0;//sud:16Apr'23--New billing Structure
    public PatientInsurancePackageId: number = null;
    public IsInsurance: boolean = false;
    public DiscountSchemeId: number = null;
    public IsCashBillSyncToAcc: boolean = false;
    public IsCreditBillSyncToAcc: boolean = false;
    public LabTypeName: string = null;
    constructor() {
        this.CreatedOn = moment().format("YYYY-MM-DD HH:mm:ss");
    }

    //below properties are used only in client side.
}

File: Code/Components/DanpheEMR.ServerModel/BillingModels/POS/BillInvoiceReturnItemsModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

/*
 File: BillInvoiceReturnItemsModel.cs
 Created: 1May'21 <sudarshan>
 Description: Invoice items for return (credit note) */


namespace DanpheEMR.ServerModel
{
    public class BillInvoiceReturnItemsModel
    {
        [Key]
        public int BillReturnItemId { get; set; }
        public int BillReturnId { get; set; }
        public int BillingTransactionItemId { get; set; }
        public int BillingTransactionId { get; set; }
        public int PatientId { get; set; }
        public int ServiceDepartmentId { get; set; }
        public int ServiceItemId { get; set; }//Sud:16Apr'23--For New BillingStructures
        public string ItemName { get; set; }
        public double Price { get; set; }
        public double RetQuantity { get; set; }
        public double RetSubTotal { get; set; }
        public double RetDiscountAmount { get; set; }
        public double RetTaxAmount { get; set; }
        public double RetTotalAmount { get; set; }
        public double RetDiscountPercent { get; set; }
        public int? PerformerId { get; set; }
        public string BillStatus { get; set; }
        public Int64? RequisitionId { get; set; }
        public DateTime? RequisitionDate { get; set; }
        public int RetCounterId { get; set; }
        public string RetRemarks { get; set; }
        public int? PrescriberId { get; set; }
        public int? PatientVisitId { get; set; }
        public int? BillingPackageId { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public string BillingType { get; set; }
        public int? RequestingDeptId { get; set; }
        public string VisitType { get; set; }
        public string PriceCategory { get; set; }//This column can be removed later..
        public int PriceCategoryId { get; set; }//Sud:16Apr'23--For New BillingStructures
        public int? PatientInsurancePackageId { get; set; }
        public bool IsInsurance { get; set; }
        public int? DiscountSchemeId { get; set; }
        public bool IsCashBillSyncToAcc { get; set; }
        public bool IsCreditBillSyncToAcc { get; set; }
        public string LabTypeName { get; set; }

    }
}
